source/client-side-encryption/client-side-encryption.rst

```diff
@@ -428,23 +443,46 @@ key. If the kmsProvider is "aws" it is required and has the following fields:
 .. code:: typescript
 
    {
-      region: String, // Required.
-      key: String, // Required. The Amazon Resource Name (ARN) to the AWS customer master key (CMK).
-      endpoint: String // Optional. An alternate host identifier to send KMS requests to. May include port number.
+      region: String,
+      key: String, // The Amazon Resource Name (ARN) to the AWS customer master key (CMK).
+      endpoint: Optional<String> // An alternate host identifier to send KMS requests to. May include port number.
    }
 
-Drivers MUST document the expected value of masterKey for "aws" and that
-it is required, not optional.
+If the kmsProvider is "azure" the masterKey is required and has the following fields:
 
-The value of "endpoint" is a host name with optional port number separated by a
+.. code:: typescript
+
+   {
+      keyVaultEndpoint: String, // Host with optional port. Example: "example.vault.azure.net".
+      keyName: String,
+      keyVersion: Optional<String>
+   }
+
+If the kmsProvider is "gcp" the masterKey is required and has the following fields:
+
+.. code:: typescript
+
+   {
+      projectId: String,
+      location: String,
+      keyRing: String,
+      keyName: String,
+      keyVersion: Optional<String>,
+      endpoint: Optional<String> // Host with optional port. Defaults to "cloudkms.googleapis.com".
+   }
+
+If the kmsProvider is "local" the masterKey is not applicable.
+
+Drivers MUST document the expected value of masterKey for "aws", "azure", and "gcp" and
```

Comment:
Instead of 'expected value' can we say something like:
```
Drivers MUST document the expected fields in the masterKey document for the "aws", "azure", and "gcp"
KMS providers. Additionally, they MUST document that masterKey is **required** for these providers and
is not optional.
```

I also suggested amending L440 above to `The masterKey document identifies a...`